neurolib/models/multimodel/builder/adex.py

```diff
@@ -0,0 +1,944 @@
+        xid = (xi - sigma_range[0]) / d_sigma
+        xid1 = np.floor(xid)
+        dxid = xid - xid1
+        yid = (yi - mu_range[0]) / d_mu
+        yid1 = np.floor(yid)
+        dyid = yid - yid1
+        return int(xid1), int(yid1), dxid, dyid
+
+    # outside one boundary
+    if yi < mu_range[0]:
+        yid1 = 0
+        dyid = 0.0
+        if xi >= sigma_range[0] and xi < sigma_range[-1]:
+            xid = (xi - sigma_range[0]) / d_sigma
+            xid1 = np.floor(xid)
+            dxid = xid - xid1
+        elif xi < sigma_range[0]:
+            xid1 = 0
+            dxid = 0.0
+        else:  # xi >= x(end)
+            xid1 = -1
+            dxid = 0.0
+        return int(xid1), int(yid1), dxid, dyid
+
+    if yi >= mu_range[-1]:
+        yid1 = -1
+        dyid = 0.0
+        if xi >= sigma_range[0] and xi < sigma_range[-1]:
+            xid = (xi - sigma_range[0]) / d_sigma
+            xid1 = np.floor(xid)
+            dxid = xid - xid1
+        elif xi < sigma_range[0]:
+            xid1 = 0
+            dxid = 0.0
+        else:  # xi >= x(end)
+            xid1 = -1
+            dxid = 0.0
+        return int(xid1), int(yid1), dxid, dyid
+
+    if xi < sigma_range[0]:
+        xid1 = 0
+        dxid = 0.0
+        yid = (yi - mu_range[0]) / d_mu
+        yid1 = np.floor(yid)
+        dyid = yid - yid1
+        return int(xid1), int(yid1), dxid, dyid
+
+    if xi >= sigma_range[-1]:
+        xid1 = -1
+        dxid = 0.0
+        yid = (yi - mu_range[0]) / d_mu
+        yid1 = np.floor(yid)
+        dyid = yid - yid1
+        return int(xid1), int(yid1), dxid, dyid
+
+
+@numba.njit()
+def _table_lookup(
+    current_mu, current_sigma, sigma_range, mu_range, d_sigma, d_mu, cascade_table,
+):
+    """
+    Translate mean and std. deviation of the current to selected quantity using
+    linear-nonlinear lookup table for AdEx. Defined as jitted function due to
+    compatibility with numba backend.
+    """
+    x_idx, y_idx, dx_idx, dy_idx = _get_interpolation_values(
+        current_sigma, current_mu, sigma_range, mu_range, d_sigma, d_mu
+    )
+    return (
+        cascade_table[y_idx, x_idx] * (1 - dx_idx) * (1 - dy_idx)
+        + cascade_table[y_idx, x_idx + 1] * dx_idx * (1 - dy_idx)
+        + cascade_table[y_idx + 1, x_idx] * (1 - dx_idx) * dy_idx
+        + cascade_table[y_idx + 1, x_idx + 1] * dx_idx * dy_idx
+    )
+
+
+class AdExMass(NeuralMass):
+    """
+    Adaptive exponential integrate-and-fire mean-field mass. Can be excitatory
+    or inhibitory, depending on the parameters.
+    """
+
+    name = "AdEx mean-field mass"
+    label = "AdExMass"
+    # define python callback function name for table lookup (linear-nonlinear
+    # approximation of Fokker-Planck equation)
+    python_callbacks = ["firing_rate_lookup", "voltage_lookup", "tau_lookup"]
+
+    num_noise_variables = 1
+
+    @staticmethod
+    def _rescale_strengths(params):
```

Comment:
ALN uses rescaled strengths so it's done here... plus `tau_m` is computed from other parameters